src/Security/Authentication/Core/src/Events/ResultContext.cs

```diff
@@ -24,26 +26,30 @@ protected ResultContext(HttpContext context, AuthenticationScheme scheme, TOptio
         /// <summary>
         /// Gets or sets the <see cref="ClaimsPrincipal"/> containing the user claims.
         /// </summary>
-        public ClaimsPrincipal Principal { get; set; }
+        public ClaimsPrincipal? Principal { get; set; }
 
-        private AuthenticationProperties _properties;
         /// <summary>
         /// Gets or sets the <see cref="AuthenticationProperties"/>.
         /// </summary>
-        public AuthenticationProperties Properties {
-            get => _properties ?? (_properties = new AuthenticationProperties());
+        public AuthenticationProperties Properties
+        {
+            get
+            {
+                _properties ??= new AuthenticationProperties();
+                return _properties;
+            }
             set => _properties = value;
         }
 
         /// <summary>
         /// Gets the <see cref="AuthenticateResult"/> result.
         /// </summary>
-        public AuthenticateResult Result { get; private set; }
+        public AuthenticateResult Result { get; private set; } = default!;
 
         /// <summary>
         /// Calls success creating a ticket with the <see cref="Principal"/> and <see cref="Properties"/>.
         /// </summary>
-        public void Success() => Result = HandleRequestResult.Success(new AuthenticationTicket(Principal, Properties, Scheme.Name));
+        public void Success() => Result = HandleRequestResult.Success(new AuthenticationTicket(Principal!, Properties, Scheme.Name));
```

Comment:
What happens if Principal is null? Ok because AuthenticationTicket checks for null?